odl/discr/diff_ops.py

```diff
@@ -815,102 +881,51 @@ def finite_diff(f, axis=0, dx=1.0, method='forward', out=None, **kwargs):
             raise ValueError('expected output shape {}, got {}'
                              ''.format(f.shape, out.shape))
 
-    if f_arr.shape[axis] == 2 and edge_order == 2:
-        raise ValueError('shape of array to small to use edge order 2')
+    if f_arr.shape[axis] == 2 and pad_mode == 'order2':
+        raise ValueError("shape of array to small to use 'order2'")
```

Comment:
Also: "shape to small" is ambiguous, could also mean that the tuple is too short.